### DISCUSSION TITLE
Why do my variables are not being pushed in my timeline?

### DISCUSSION THREAD

**Discussion by Ricardo-Piccolo** [jspsych/jsPsych#3027] [Q&A] [answered]
Hey y'all! I'd like some help, if possible, with a simple experiment. It is supposed to be a go/no-go experiment in the future, but i'm still at the very beginning of it. I can't push my second test (named as 'test2') in my timeline. What should I do?

<!DOCTYPE html>
<html>
    <head>
        <title>Image_Trial</title>
        <script src="jsPsych/jspsych.js"></script>
        <script src="jsPsych/plugins/jspsych-html-keyboard-response.js"></script>
        <script src="jsPsych/plugins/jspsych-image-keyboard-response.js"></script>
        <script src="jsPsych/plugins/jspsych-instructions.js"></script>
        <link href="jsPsych/css/jspsych.css" rel="stylesheet">
    </head>
    <body></body>
    <script>

        var timeline = [];

        var instructions = {
            type: 'instructions',
            pages: ['<p>Bem-vindo(a) à esse experimento!</p><p>Quadrados de diferentes cores aparecerão na tela.</p><p>Quando o quadrado for azul, aperte a tecla ESPAÇO no teclado o mais rápido que puder.</p><p>Se o quadrado for de outra cor, não pressione nenhuma tecla.</p><p>Você fará um teste agora para melhor compreender as instruções.</p><p>Aperte o NEXT para dar início ao teste.</p>'],
            show_clickable_nav: true
        }
        timeline.push(instructions);

        var test1 = {
        type: 'image-keyboard-response',
        stimulus: ['bluesquare.png'],
        choices: ' ',
        stimulus_duration: 1000,
        post_trial_gap: 200,
        response_ends_trial: false
        }
        timeline.push(test1);

        var test2 = {
        type: 'image-keyboard-response',
        stimulus: ['yellowsquare.png'],
        choices: ' ',
        stimulus_duration: 1000,
        post_trial_gap: 200,
        response_ends_trial: false
        }
        timeline.push(test2);

        jsPsych.init({
            timeline: [instructions, test1, test2],
            on_finish: function() {
                jsPsych.data.displayData();
            }
        })
    </script>
</html>

**Comment by jodeleeuw** [comment, marked as answer]
Hi @Ricardo-Piccolo,

The `test1` trial is currently configured in a way that prevents it from ending. `response_ends_trial: true` will make it so that no response can end the trial, and the default value for `trial_duration` is infinite. If you want the trial to end after 1 second, set `trial_duration: 1000` instead of `stimulus_duration`.
